### DISCUSSION TITLE
How do I use mocks on prod mode?

### DISCUSSION THREAD

**Discussion by ShookLyngs** [pengzhanbo/vite-plugin-mock-dev-server#25] [Q&A] [answered]
Is it possible to use the mock-dev-server features in production mode?

**Comment by pengzhanbo** [comment, marked as answer]
The plugin only supports running under `vite server` and `vite preview` mode by default.

And because mock files support importing node modules and third-party modules, the mock code will not be packaged into the project source code.

However, it does support packaging the mock files as a small independent Node service that can be deployed separately in `vite build` mode. See `options.build` in [configuration documentation](https://github.com/pengzhanbo/vite-plugin-mock-dev-server#mockdevserverpluginoptions) and [Mock Services](https://github.com/pengzhanbo/vite-plugin-mock-dev-server#mock-services).

You can deploy this Node service separately and then use an HTTP service such as Nginx for proxy according to your application scenario so that frontend projects can use these mock interfaces.

```ts
import { defineConfig } from 'vite'
import mockDevServerPlugin from 'vite-plugin-mock-dev-server'

export default defineConfig({
  plugins: [
    mockDevServerPlugin({
      build: true,
    }),
  ]
})
```
